### PR TITLE
Ensure that all server managed triples are suppressed on omit.

### PR DESCRIPTION
**Pull request by mohideen**
**JIRA Ticket**: 
- https://jira.duraspace.org/browse/FCREPO-2877
- https://jira.duraspace.org/browse/FCREPO-2880

# What does this Pull Request do?
Ensure that all server managed triples are suppressed in RDF body when omitting Server Managed Triples

# How should this be tested?
1. Verify that all the relevant SMTs are being tested for in FedoraLdpIT
- testGetObjectOmitMembership
- testGetObjectOmitContainment
- testGetObjectOmitServerManaged
2. Run the test to verify all the tests pass.

## Manual Test
- Follow steps to reproduce the issue as per the description in FCREPO-2877. 
- Verify that the `ldp:BasicContainer` is no longer being returned.

# Interested parties
@dbernstein, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -229,7 +230,8 @@
     protected ExternalContentHandlerFactory extContentHandlerFactory;
 
     private static final Predicate<Triple> IS_MANAGED_TYPE = t -> t.getPredicate().equals(type.asNode()) &&
-            isManagedNamespace.test(t.getObject().getNameSpace());
+            (isManagedNamespace.test(t.getObject().getNameSpace()) ||
+            isManagedLDPType.test(createResource(t.getObject().getURI())));
```
Comment (whikloj) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Just wondering if this should be removed and replaced with the `isManagedPredicate` from `RdfLexicon` which contains other stuff as well?
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java#L280-L281

Comment (whikloj) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Also we shouldn't be suppressing `<> rdf:type ldp:BasicContainer` when we suppress SMTs. This is not an SMT, I think

Comment (awoods) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: The "Are LDP Container types SMTs" question should be resolved on the 2018-10-04 Tech call.

As for the `isManagedPredicate`, the test here is:
- is the predicate `rdf:type`, and
- is the object in a managed namespace, or
- is the object a managed LDP type

@whikloj : are you suggesting the possibility of changing the logic to:
- is the predicate `rdf:type`, and
- is the predicate managed?

Comment (whikloj) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: I am just unclear as to what is an SMT and that makes testing for them very difficult. I think I should just wait until that is decided then I can have an opinion.

Comment (whikloj) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: So following the discussion on the tech call my opinion is that the existing `isManagedNamespace` predicate could be expand to include both `fedora:*` and `ldp:*` and then this change to add `isManagedType` would be unnecessary. Was that your understand as well @mohideen @awoods?

Comment (awoods) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: @whikloj : That sounds right based on today's call.

Comment (mohideen) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: @whikloj @awoods So, we won't be able to create resources with `ldp:*` rdf:type?

```
$ curl -i -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/testputldp -X PUT -H "Content-Type: text/turtle" --data-binary '<> a <http://www.w3.org/ns/ldp#DirectContainer> .'
HTTP/1.1 409 Conflict
Date: Thu, 04 Oct 2018 23:44:06 GMT
Set-Cookie: JSESSIONID=186v9jsw7ckum1qnzffgc3lp6s;Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Wed, 03-Oct-2018 23:44:06 GMT
Link: <http://localhost:8080/static/constraints/ServerManagedTypeException.rdf>; rel="http://www.w3.org/ns/ldp#constrainedBy"
Content-Type: text/plain;charset=utf-8
Content-Length: 98
Server: Jetty(9.3.1.v20150714)

The repository type (http://www.w3.org/ns/ldp#DirectContainer) of this resource is system managed.
```

Did we say we will use `Link` headers to overcome this?

Comment (mohideen) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Refactored based on today's decisions.

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 2]
```diff
@@ -2142,11 +2267,11 @@ public void testPutToChangeInteractionModelWithRdf() throws IOException {
         assertTrue(getLinkHeaders(getObjMethod(pid + "/b")).contains(DIRECT_CONTAINER_LINK_HEADER));
 
         // successful update the properties with the interaction mode
-        final String ttla = "<> a <" + DIRECT_CONTAINER.getURI()
-                + "> ; <" + MEMBERSHIP_RESOURCE + "> <" + container + ">;\n"
+        final String ttla = "<> <" + MEMBERSHIP_RESOURCE + "> <" + container + ">;\n"
                 + "<" + HAS_MEMBER_RELATION + "> <info:some/relation> .\n";
         final HttpPut puta = putObjMethod(pid + "/b", "text/turtle", ttla);
         puta.addHeader("Prefer", "handling=lenient; received=\"minimal\"");
+        puta.addHeader(LINK, "<http://www.w3.org/ns/ldp#DirectContainer>;rel=type");
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java]: Maybe replace this String with the constant: `DIRECT_CONTAINER_LINK_HEADER`

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 3]
```diff
@@ -2256,8 +2381,8 @@ public void testGetObjectReferencesIndirect() throws Exception {
         // create an indirect container
         final HttpPut createContainer = new HttpPut(serverAddress + pid1 + "/members");
         createContainer.addHeader(CONTENT_TYPE, "text/turtle");
-        final String membersRDF = "<> a <http://www.w3.org/ns/ldp#IndirectContainer>; "
-            + "<http://www.w3.org/ns/ldp#hasMemberRelation> <" + memberRelation + ">; "
+        createContainer.addHeader(LINK, "<http://www.w3.org/ns/ldp#IndirectContainer>;rel=type");
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java]: Maybe replace this String with the constant: `INDIRECT_CONTAINER_LINK_HEADER`

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 4]
```diff
@@ -3640,8 +3765,8 @@ public void testContainerLastModified() throws Exception {
         final long lastmod3;
         final HttpPut createContainer = new HttpPut(objURI + "/members");
         createContainer.addHeader(CONTENT_TYPE, "text/turtle");
-        final String membersRDF = "<> a <http://www.w3.org/ns/ldp#DirectContainer>; "
-            + "<http://www.w3.org/ns/ldp#hasMemberRelation> <http://pcdm.org/models#hasMember>; "
+        createContainer.addHeader(LINK, "<http://www.w3.org/ns/ldp#DirectContainer>;rel=type");
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java]: Maybe replace this String with the constant: `DIRECT_CONTAINER_LINK_HEADER`

**Pull request merged**